### PR TITLE
<AddressInput/> - update to match design spec

### PR DESCRIPTION
**Pull request by gendeld**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: src/AddressInput/AddressInput.js [hunk 1]
```diff
@@ -3,40 +3,37 @@ import PropTypes from 'prop-types';
...
-    const { onChange } = this.props;
-    onChange && onChange(value);
-  };
-
-  _debouncedOnChange = debounce(this._onChange, this.props.debounceDuration, {
-    leading: true,
-  });
-
-  _onInputChange = ({ target: { value } }) => {
+  _onChange = event => {
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:22]: Debouncing is being implemented at the hook level

File: src/AddressInput/AddressInput.js [hunk 2]
```diff
@@ -51,30 +48,102 @@ class AddressInput extends React.PureComponent {
     );
   };
 
-  render() {
+  _getInputValue = () => {
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:51]: Allow passing `value` prop to have controlled input

File: src/AddressInput/AddressInput.js [hunk 3]
```diff
@@ -51,30 +48,102 @@ class AddressInput extends React.PureComponent {
...
+
+    return {
+      id: '',
+      disabled: true,
+      overrideOptionStyle: !isString,
+      value,
+    };
+  };
+
+  _renderOptions = () => {
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:99]: Add Empty and Loading states to component

File: src/AddressInput/AddressInput.js [hunk 4]
```diff
@@ -51,30 +48,102 @@ class AddressInput extends React.PureComponent {
...
-      status,
+      placeholder,
+      disabled,
     } = this.props;
+    const { isDropdownOpen } = this.state;
+    const value = this._getInputValue();
+
+    const isLoading = this._getIsLoading();
+
+    const showLoadingStatusIndicator = isLoading && !isDropdownOpen;
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:130]: In spec, when dropdown is closed - loader is shown on the input,
and when it is open, loader is shown in the dropdown instead

File: src/AddressInput/AddressInput.js [hunk 5]
```diff
@@ -83,6 +152,9 @@ class AddressInput extends React.PureComponent {
           </Input.IconAffix>
         }
         roundInput={roundInput}
+        placeholder={placeholder}
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:155]: Add placeholder

File: src/AddressInput/AddressInput.js [hunk 6]
```diff
@@ -119,19 +194,19 @@ AddressInput.propTypes = {
   onClear: PropTypes.func,
 
   /** Shows a status indication, will mostly be used for “loading” indication upon async request calls. */
-  status: PropTypes.oneOf(['loading', 'error', 'warning']),
+  status: PropTypes.oneOf(['loading']),
```
Comment (gendeld) [inline src/AddressInput/AddressInput.js:197]: Currently there are no `error` or `warning` states described in the spec

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6400

File: src/AddressInput/AddressInput.js [hunk 7]
```diff
@@ -3,40 +3,37 @@ import PropTypes from 'prop-types';
...
-
-  _onInputChange = ({ target: { value } }) => {
+  _onChange = event => {
     this.setState({
-      inputValue: value,
+      inputValue: event.target.value,
     });
-    this._debouncedOnChange(value);
+    if (this.props.onChange) {
+      this.props.onChange(event);
```
Comment (EgoziE) [inline src/AddressInput/AddressInput.js:27]: For readability only destruct onChange from this.props

File: src/AddressInput/AddressInput.js (hunk 1 above)
Comment (EgoziE) [reply src/AddressInput/AddressInput.js:22]: k

File: src/AddressInput/AddressInput.js [hunk 8]
```diff
@@ -51,30 +48,102 @@ class AddressInput extends React.PureComponent {
...
+      overrideOptionStyle: !isString,
+      value,
+    };
+  };
+
+  _renderOptions = () => {
+    const { options } = this.props;
+    const value = this._getInputValue();
+    const isLoading = this._getIsLoading();
+    const noResultsFound = !options || !options.length;
```
Comment (EgoziE) [inline src/AddressInput/AddressInput.js:103]: Just for readability:
```suggestion
    const noResultsFound = !options || options.length === 0;
```

File: src/AddressInput/AddressInput.js (hunk 4 above)
Comment (EgoziE) [reply src/AddressInput/AddressInput.js:130]: Sound like something that should be mentioned as a comment for future possible questions.

File: src/AddressInput/AddressInput.js (hunk 6 above)
Comment (EgoziE) [reply src/AddressInput/AddressInput.js:197]: I know that it's not described directly in Address Input page, but this component is part of the Input family and "status" is a basic behavior of IC.Input family.
Therefore I wouldn't remove it.

File: src/AddressInput/AddressInput.js [hunk 9]
```diff
@@ -140,6 +215,8 @@ AddressInput.defaultProps = {
   roundInput: true,
   optionsLayout: 'single-line',
   showOptionsIcons: true,
+  size: 'medium',
+  noResultsText: 'No results found',
```
Comment (EgoziE) [inline src/AddressInput/AddressInput.js:219]: This is not a good idea in my opinion.
It assumes that English is the default language, let's talk with UX, if they want this prop to have a value no matter what, maybe we should make it mandatory.

File: src/AddressInput/index.d.ts [hunk 10]
```diff
@@ -9,16 +9,17 @@ export interface AddressInputProps {
...
-  onSelect?: (value: DropdownLayoutValueOption) => void;
-  debounceDuration?: number;
+  value?: string;
+  disabled?: boolean;
+  onSelect?: (option: DropdownLayoutValueOption) => void;
   onChange?: React.ChangeEventHandler<HTMLInputElement>;
   options?: DropdownLayoutOption[];
   onClear?: () => void;
-  status?: 'loading' | 'error' | 'warning';
+  status?: 'loading';
```
Comment (EgoziE) [inline src/AddressInput/index.d.ts:18]: same comment

File: src/AddressInput/index.d.ts [hunk 11]
```diff
@@ -9,16 +9,17 @@ export interface AddressInputProps {
...
   roundInput?: boolean;
-  optionsLayout?: 'single-line' | 'double-line';
-  showOptionsIcons?: boolean;
   size?: 'small' | 'medium' | 'large';
+  placeholder?: string;
+  noResultsText?: string | React.ReactNode;
```
Comment (EgoziE) [inline src/AddressInput/index.d.ts:22]: String is a sub  type of React.ReactNode
```suggestion
  noResultsText?: React.ReactNode;
```

**Review by EgoziE — CHANGES_REQUESTED** [review]

**Review by EgoziE — APPROVED** [review]

**Comment by assafzi** [comment]
UX here.
Looks great.
only 2 things to improve:

- [ ] Add, at the description that this component uses <AddressInputItem/>, like this:
image

- [ ] Loader should be before the clear button (now it's after)
image

**Comment by assafzi** [comment]
Approved by UX
Great descriptions!
Thanks

**Pull request merged**